Manito/Manito/Presentation/Scene/DetailScene/Detail/ViewModel/MemoryViewModel.swift

```diff
@@ -0,0 +1,99 @@
+//
+//  MemoryViewModel.swift
+//  Manito
+//
+//  Created by SHIN YOON AH on 10/16/23.
+//
+
+import Combine
+import Foundation
+
+final class MemoryViewModel: BaseViewModelType {
+
+    enum MemberType: Int {
+        case manitte = 0
+        case manitto = 1
+    }
+
+    struct Input {
+        let viewDidLoad: AnyPublisher<Void, Never>
+        let segmentControlValueChanged: AnyPublisher<Int, Never>
+    }
+
+    struct Output {
+        let member: AnyPublisher<Result<(announcingText: String, memberInfo: MemberInfo), Error>, Never>
+        let messages: AnyPublisher<Result<[MessageListItem], Error>, Never>
+    }
+
+    // MARK: - property
+    
+    private let memberSubject: PassthroughSubject<Result<(announcingText: String, memberInfo: MemberInfo), Error>, Never> = PassthroughSubject()
+    private let messageSubject: PassthroughSubject<Result<[MessageListItem], Error>, Never> = PassthroughSubject()
+
+    private var cancelBag: Set<AnyCancellable> = Set()
+
+    private let usecase: MemoryUsecase
+    private let roomId: String
+
+    // MARK: - init
+
+    init(usecase: MemoryUsecase,
+         roomId: String) {
+        self.usecase = usecase
+        self.roomId = roomId
+    }
+
+    // MARK: - Public - func
+
+    func transform(from input: Input) -> Output {
+        input.viewDidLoad
+            .sink(receiveValue: { [weak self] in
+                self?.fetchMemory()
+            })
+            .store(in: &self.cancelBag)
+        
+        input.segmentControlValueChanged
+            .compactMap { MemberType(rawValue: $0) }
+            .sink(receiveValue: { [weak self] in
+                self?.sendInformation(with: $0)
+            })
+            .store(in: &self.cancelBag)
+
+        return Output(
+            member: self.memberSubject.eraseToAnyPublisher(),
+            messages: self.messageSubject.eraseToAnyPublisher()
+        )
+    }
+
+    // MARK: - Private - func
+
+    private func fetchMemory() {
+        Task {
+            do {
+                let _ = try await self.usecase.fetchMemory(roomId: self.roomId)
```

Comment:
p4;

```swift
func fetchMemory(roomId: String) async throws
```

해당 함수가 리턴값이 없어서 `let _`을 지워도 될 것 같아요😁
```suggestion
                try await self.usecase.fetchMemory(roomId: self.roomId)
```

Reply:
[9e672dd](https://github.com/DeveloperAcademy-POSTECH/MC2-Team5-Firefighter/pull/564/commits/9e672ddc3dc5860dae07448f3d7f778022bb3fb2)
삭제 했습니다.🙇‍♂️